### PR TITLE
AK: Free function for getting `.size()` from container

### PR DESCRIPTION
**Pull request by ldm5180**
Problem:                                                           
- `AK::array_size(T)` only works for C-arrays.                     
- `array_size` is a misleading name since it is really the size of any
  container which provides `size()` or the number of elements in a    
  C-array.                         
                                                                   
Solution:                                                          
- Add an overload so that all objects which have a `.size()` member
  function can be called using the free function `array_size(T)`.  
- Change to just be `size()` so that it is a consistent name for      
  different types.

### CODE REVIEW

**Comment by ldm5180** [comment]
I don't love the name `size()` here. I don't have anything better. `container_size()` or `data_size()` or `num_elements()`? Yuck.

**Comment by tuckerpo** [comment]
> I don't love the name `size()` here. I don't have anything better. `container_size()` or `data_size()` or `num_elements()`? Yuck.

IMO `num_elements()` is much more clear than `size(T)` now that it works for many container types.
In a systems programming context, my brain defaults to assuming that `size(T)` refers to size in bytes, not number of elements that T contains (which, sometimes, is size in bytes too).

File: AK/StdLibExtras.h [hunk 1]
```diff
@@ -68,6 +68,13 @@ constexpr SizeType array_size(T (&)[N])
     return N;
 }
 
+template<typename T>
+requires requires { Detail::Decay<T> {}.size(); }
```
Comment (kleinesfilmroellchen) [inline AK/StdLibExtras.h]: Why are there two `requires` here?

**Review by kleinesfilmroellchen — COMMENTED** [review]
I think the name is fine, as we're mirroring the size member method that already exists on all array-like types (see #12162). We can have a discussion all day about whether `length`, `len`, `size`, `count`, or whatever is the best name, but it doesn't really matter. All of these would be fine, so let's stick with what we have.

File: AK/StdLibExtras.h (hunk 1 above)
Comment (ldm5180) [reply AK/StdLibExtras.h]: `requires requires`
https://quuxplusone.github.io/blog/2019/01/15/requires-requires-is-like-noexcept-noexcept/

Comment (kleinesfilmroellchen) [reply AK/StdLibExtras.h]: Thanks, C++ is ridiculous sometimes but 🤷

**Pull request closed without merging**